docs/design/net-topology-aware.md

```diff
@@ -0,0 +1,90 @@
+# Net Topology Aware Plugin
+
+- [Net Topology Aware Plugin](#net-topology-aware-plugin)
+  - [Backgrounds](#backgrounds)
+  - [Motivation](#motivation)
+  - [Proposal one](#proposal-one)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+    - [Design Action](#design-action)
+      - [Pod scheduling process](#pod-scheduling-process)
+      - [Usage](#usage)
+    - [Drawbacks](#drawbacks)
+
+## Backgrounds
+
+A Kubernetes cluster typically comprises numerous nodes distributed across different IDCs, chassis, and switches.
+
+Data transformations vary in performance across these different components.
+
+For latency-sensitive workloads, it's crucial to execute tasks within the same IDC and ideally on the same chassis and switch.
+
+## Motivation
+
+The goal is to make the Kubernetes scheduler network-topology aware to achieve the following:
+
+Ensure optimal scheduling of tasks from the same job onto nodes within the same topology, such as the same IDC, chassis, or switch.
+
+There will be two types of network-topology aware
+
+- **static**: `net-topology.type: static` is aiming to aware the network topology by nodes' labels
+- **dynamic**: `net-topology.type: dynamic` is aiming to use some tools to detect the network topology dynamically
```

Comment:
Is there any detail for network topology discovery?

Reply:
For example, `ibnetdiscover` can be used to discover the InfiniBand network topology.
I have add an example tool in docs.